Autopilot/Path Manager/PathManager.c

```diff
@@ -667,12 +667,37 @@ void copyGPSData(){
     pmData.checkbyteDMA2 = generatePMDataDMAChecksum2();
 }
 
+// check if power on reset, brown out reset, watchdog time out or software reset flags are set
+
+ char checkForReset(){
+     if (RCONbits.POR || RCONbits.BOR || RCONbits.WDTO || RCONbits.SWR || RCONbits.EXTR){
+         // clear RCON(Reset control) bits)
+         RCONbits.BOR = 0;
+         RCONbits.EXTR = 0;
+         RCONbits.POR = 0;
+         RCONbits.SWR = 0;
+         RCONbits.WDTO = 0;
+         return 1;
+     }
+     else 
+         return 0;
+ } 
+ void GPSLock(){
+     if(checkForReset()){
```

Comment:
No need for the checkForReset function at all, since its only ever going to be used once. I'd just call this function `checkFirstGPSLock` or something, check the startup error code registers, and then set the gps coordinates.